packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -118,12 +118,31 @@ export class MediaPlaceholder extends Component {
 			onError,
 			onSelect,
 			value = [],
+			dynamicAlteration,
 		} = this.props;
 		let setMedia;
 		if ( multiple ) {
 			if ( addToGallery ) {
-				const currentValue = value;
+				let currentValue = value;
 				setMedia = ( newMedia ) => {
+					if ( dynamicAlteration ) {
+						const galleryValue = this.props.value || [];
+						// If currentValue contains things not in galleryValue, remove them.
+						currentValue = currentValue.filter( ( item ) => {
+							return galleryValue.some( ( propItem ) => propItem.id === item.id && propItem.url === item.url );
```

Comment:
One thing I found confusing is the naming `currentValue` vs `galleryValue`. Would it make more sense to say something like `incomingValue`?

Reply:
or newValues or something like that

Reply:
I agree!  We should probably update this, I had been doing some debating about how to properly rename these but ended up sending myself in circles.

Essentially, we should rename `galleryValue` to be `currentValue`, as it does represent the current value that is in place in the gallery block.  Then `currentValue` could be named something like `initialValue`, since it is initially defined at the start of the upload.  But then `initialValue` seems to be a bad name for it since we are updating its contents when the function runs to no longer be the initial value... LOL..

What about renaming `galleryValue` to `currentValue`, and rename `currentValue` as `returnValue` ?  Even that name has some issues since `newMedia` is `.concat()`d onto that at the end, so it isn't really the full return value...  I am open for suggestions on what to rename `currentValue` to if you have any ideas?

Reply:
Hm -- I feel like I still get confused by that. It may be worth not saying `Value`, but saying `Images` or `Uploads`. Just something to nail down exactly what we're talking about. Something like `currentUploads` or `currentImages`. I feel like `returnValues` is confusing, but maybe more along that line of thinking like `finalUploads`.

Is the difference here something like `pendingUploads` (for `galleryValue`) and then `completedUploads`? that would be a lot clearer to me, but I'm not sure if that's actually what the code is doing

Reply:
> that would be a lot clearer to me, but I'm not sure if that's actually what the code is doing

Thats not what its doing unfortunately, but I think something like `currentMedia` and `mediaToReturn` would be a bit better than what is currently there?  `currentMedia` would be as we see the state of `this.props.value` when the function runs, and `mediaToReturn` would initially be assigned as `value` and updated along the lifecycle of the function.